### PR TITLE
workflows: set `HOMEBREW_NO_BUILD_ERROR_ISSUES` more consistently

### PR DESCRIPTION
**Pull request by carlocab**
Let's do this for every workflow where we build something.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -14,6 +14,7 @@ env:
   HOMEBREW_NO_INSTALL_FROM_API: 1
   HOMEBREW_TEST_BOT_ANALYTICS: 1
   HOMEBREW_ENFORCE_SBOM: 1
+  HOMEBREW_NO_BUILD_ERROR_ISSUES: 1
```
Comment (Bo98) [inline .github/workflows/tests.yml:17]: Thoughts on setting this in test bot instead?

**Review by Bo98 — APPROVED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (carlocab) [reply .github/workflows/tests.yml:17]: Wouldn't that enforce it for third-party taps too? Do we want that?

Comment (Bo98) [reply .github/workflows/tests.yml:17]: Is it useful information in the context of `brew test-bot`? I understand its use for end users but `brew test-bot` is a different audience.

**Comment by MikeMcQuaid** [comment]
Makes sense, thanks @carlocab!

I have a dream that one day "all the workflows where we build something" would be a single workflow...

**Comment by carlocab** [comment]
> I have a dream that one day "all the workflows where we build something" would be a single workflow...

One is a tall order. Two seems doable.

**Comment by MikeMcQuaid** [comment]
@carlocab in my mind the `tests` workflow could take the combined inputs of the dispatch workflow(s) and adjust behaviour accordingly.

**Pull request merged**